### PR TITLE
[Fixes #19666] [Components] Improve reliability of component quarantined tests (take 2)

### LINKED ISSUE #19666: ComponentHubReliabilityTest are unreliable
```
Microsoft.AspNetCore.SignalR.HubException : Unable to complete handshake with the server due to an error: Handshake was canceled.
  at Microsoft.AspNetCore.SignalR.Client.HubConnection.HandshakeAsync(ConnectionState startingConnectionState, CancellationToken cancellationToken) in /_/src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs:line 1057
   at Microsoft.AspNetCore.SignalR.Client.HubConnection.StartAsyncCore(CancellationToken cancellationToken) in /_/src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs:line 433
   at Microsoft.AspNetCore.SignalR.Client.HubConnection.StartAsyncCore(CancellationToken cancellationToken) in /_/src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs:line 441
   at Microsoft.AspNetCore.SignalR.Client.HubConnection.StartAsyncInner(CancellationToken cancellationToken) in /_/src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs:line 255
   at System.Threading.Tasks.ForceAsyncAwaiter.GetResult() in /_/src/SignalR/common/Shared/ForceAsyncAwaiter.cs:line 40
   at Microsoft.AspNetCore.SignalR.Client.HubConnection.StartAsync(CancellationToken cancellationToken) in /_/src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs:line 232
   at Ignitor.BlazorClient.ConnectAsync(Uri uri, Boolean connectAutomatically) in /_/src/Components/Ignitor/src/BlazorClient.cs:line 359
   at Microsoft.AspNetCore.Components.E2ETest.ServerExecutionTests.ComponentHubReliabilityTest.ComponentLifecycleMethodThrowsExceptionTerminatesTheCircuit(String id) in /_/src/Components/test/E2ETest/ServerExecutionTests/ComponentHubReliabilityTest.cs:line 295
--- End of stack trace from previous location -
```

### PR DESCRIPTION
**Pull request by javiercn**
* Tries to increase the reliability of the tests by:
  * Trying to ensure that the server is up and running before connecting.
  * Retrying a connection attempt multiple times.

### CODE REVIEW

File: src/Components/Ignitor/src/BlazorClient.cs [hunk 1]
```diff
@@ -359,7 +359,19 @@ public async Task<bool> ConnectAsync(Uri uri, bool connectAutomatically = true,
             HubConnection.On<string>("JS.Error", OnError);
             HubConnection.Closed += OnClosedAsync;
 
-            await HubConnection.StartAsync(CancellationToken);
+            for (var i = 0; i < 10; i++)
+            {
+                try
+                {
+                    await HubConnection.StartAsync(CancellationToken);
```
Comment (javiercn) [inline src/Components/Ignitor/src/BlazorClient.cs:366]: @BrennanConroy will this leave the hubconnection in an invalid state if the call fails?

Comment (BrennanConroy) [reply src/Components/Ignitor/src/BlazorClient.cs:366]: It will leave it in a disconnected state

Comment (javiercn) [reply src/Components/Ignitor/src/BlazorClient.cs:366]: I'm explictly asking if re-using the same hubconnection for the retries is OK

Comment (BrennanConroy) [reply src/Components/Ignitor/src/BlazorClient.cs:366]: Yes.

Although I'd love to know why you think this is needed. SignalR has 100's of tests that use the HubConnection to connect to a server and doesn't need to use retries.

Comment (javiercn) [reply src/Components/Ignitor/src/BlazorClient.cs:366]: I'm just being cautious here. Based on the error I have not much idea of what can be going wrong. Other than the server is taking longer to start?

**Comment by Pilchie** [comment]
:eyes:

**Comment by BrennanConroy** [comment]
Do you have any failures I can take a quick look at?

**Comment by javiercn** [comment]
> Do you have any failures I can take a quick look at?

https://github.com/dotnet/aspnetcore/issues/19666

**Comment by javiercn** [comment]
I believe the other potential cause is the server taking longer to start on the CI, I ran the tests locally on my box 200+ times in a loop and wasn't able to repro it.

**Comment by BrennanConroy** [comment]
One interesting thing to note is that when these tests are quarantined, they don't seem to fail anymore. But when you accidentally un-quarantined them last week they failed pretty quickly.

Also, it would be nice if you could enable timestamps in the server logs. Only the client logs have timestamps currently.

**Comment by javiercn** [comment]
> One interesting thing to note is that when these tests are quarantined, they don't seem to fail anymore. But when you accidentally un-quarantined them last week they failed pretty quickly.

Bad luck? Karma? Destiny? Murphy's law?

> Also, it would be nice if you could enable timestamps in the server logs. Only the client logs have timestamps currently.

No idea how to enable that, do you have any thoughts? If it's simple I'm happy to do it.

**Comment by BrennanConroy** [comment]
> Bad luck? Karma? Destiny? Murphy's law?

I'm thinking more like machine resource starvation from the parallel test runs.

> No idea how to enable that, do you have any thoughts?

Hmm, none right now, might need to be added to the test logger. But something for future improvement.

**Comment by javiercn** [comment]
> I'm thinking more like machine resource starvation from the parallel test runs.

I didn't know we ran works in parallel. Are you thinking about thread starvation here? I believe we run the server on a separate thread (at the very least)

**Comment by BrennanConroy** [comment]
We run everything in parallel I believe, so other test projects could be running, even builds and packs could be running at the same time I think.

I think I told Pranav once that you might see improved test reliability once Components moved to Helix because then you'll have a dedicated machine for just Components tests.

**Comment by javiercn** [comment]
> We run everything in parallel I believe, so other test projects could be running, even builds and packs could be running at the same time I think.
> 
> I think I told Pranav once that you might see improved test reliability once Components moved to Helix because then you'll have a dedicated machine for just Components tests.

I thought we had a separate test group for components. That used to be the case

**Review by pranavkm — APPROVED** [review]

**Pull request merged**